src/update_agent/mod.rs

```diff
@@ -38,16 +47,31 @@ lazy_static::lazy_static! {
         "zincati_update_agent_updates_enabled",
         "Whether auto-updates logic is enabled."
     )).unwrap();
+    static ref POSTPONED_FINALIZATIONS: IntCounter = register_int_counter!(opts!(
+        "zincati_update_agent_postponed_finalizations_total",
+        "Total number of update finalization postponements due to active users."
+    )).unwrap();
+    static ref DETECTED_ACTIVE_USERS: IntGauge = register_int_gauge!(opts!(
+        "zincati_update_agent_detected_active_users",
```

Comment:
Minor suggestion: something like `finalization_detected_active_users` makes it easier to understand when this is relevant.